code/modules/power/singularity/emitter.dm

```diff
@@ -401,32 +422,33 @@
 	auto.Remove(buckled_mob)
 	. = ..()
 
-/obj/machinery/power/emitter/prototype/user_buckle_mob(mob/living/M, mob/user, check_loc = TRUE)
+/obj/machinery/power/emitter/prototype/user_buckle_mob(mob/living/mob, mob/user, check_loc = TRUE)
 	if(user.incapacitated() || !istype(user))
 		return
-	for(var/atom/movable/A in get_turf(src))
-		if(A.density && (A != src && A != M))
+	for(var/atom/movable/atom in get_turf(src))
+		if(atom.density && (atom != src && atom != mob))
 			return
-	M.forceMove(get_turf(src))
+	mob.forceMove(get_turf(src))
 	..()
 	playsound(src,'sound/mecha/mechmove01.ogg', 50, TRUE)
-	M.pixel_y = 14
+	mob.pixel_y = 14
 	layer = 4.1
-	if(M.client)
-		M.client.view_size.setTo(view_range)
+	if(mob.client)
+		mob.client.view_size.setTo(view_range)
 	if(!auto)
 		auto = new()
-	auto.Grant(M, src)
+	auto.Grant(mob, src)
 
 /datum/action/innate/protoemitter
 	check_flags = AB_CHECK_HANDS_BLOCKED | AB_CHECK_IMMOBILE | AB_CHECK_CONSCIOUS
-	var/obj/machinery/power/emitter/prototype/PE
-	var/mob/living/carbon/U
-
-
-/datum/action/innate/protoemitter/Grant(mob/living/carbon/L, obj/machinery/power/emitter/prototype/proto)
-	PE = proto
-	U = L
+	///Stores the emitter the user is currently buckled on
+	var/obj/machinery/power/emitter/prototype/proto_emitter
+	///Stores the mob instance that is buckled to the emitter
+	var/mob/living/carbon/buckled_mob
```

Comment:
Do these vars get cleared when the action is removed/qdel'd?

Reply:
i think? how can i check? or should i just put them into Destroy() and qdel them there?

Reply:
actually I was thinking about the problem incorrectly, just make sure they're nulled in Destroy and it won't cause problems ||I hope||

Reply:
done